src/components/Api.js

```diff
@@ -0,0 +1,137 @@
+import { profileTitle } from "../utils/constants";
+
+class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  //  Get User Info
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "GET",
+      headers: this._headers,
+    }).then((res) => {
```

Comment:
Needs correction: The code for processing the response is duplicated in all of the api methods. You should create a separate method `_processResponse` and use it instead. 

Your api methods will look like the following:

```js
return fetch(...).then(this._processResponse)
```

Reply:
Corrected